### PR TITLE
More volatile cli

### PR DESCRIPTION
**Pull request by PicoCentauri**
* Add more options for Universe creation
* Add a version option
* Add first simple test for argument parsing

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -56,3 +58,32 @@ def test_setup_clients(opt, dest, val):
         args = setup_clients().parse_args()
         t = type(val)
         assert t(getattr(args, dest)) == val
+
+
+class Test_analyze_data(object):
```
Comment (joaomcteixeira) [inline tests/test_cli.py:63]: Good addition.
Quick tip from my experience. I may be wrong, but I found no reason to use classes in pytest. If you use just functions and fixtures it works like charm. :coffee: talk though :wink:

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        subprocess.check_call(['mdacli'])
+
+
+def test_wrong_module():
+    """Test for a non existent module."""
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.check_call(['mdacli', 'foo'])
+
+
+@pytest.mark.parametrize('args', ("version", "help"))
```
Comment (joaomcteixeira) [inline tests/test_cli.py:32]: Interesting... I never thought of testing the actual call to the command line. That is good. Keep in mind this works because of the way `tox` installs the package. Which is good :wink:

File: tests/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    testargs = ["mdacli", "RMSF", opt]
+    if type(val) == list:
+        for i in val:
+            testargs.append(str(i))
+    else:
+        testargs.append(str(val))
+    with patch.object(sys, 'argv', testargs):
+        args = setup_clients().parse_args()
+        t = type(val)
+        assert t(getattr(args, dest)) == val
```
Comment (joaomcteixeira) [inline tests/test_cli.py:60]: You and I have found very different ways to test argparse. Very curious. Yours look great.
If you want to check mine here it is: https://github.com/joaomcteixeira/taurenmd/blob/master/tests/test_clis.py
(specially at the end of the file)

**Comment by joaomcteixeira** [comment]
:rocket:

File: tests/test_cli.py (hunk 1 above)
Comment (PicoCentauri) [reply tests/test_cli.py]: I removed the class here. However for the `Test_run_analsis` I would just keep it for better visibility :)

File: tests/test_cli.py (hunk 2 above)
Comment (PicoCentauri) [reply tests/test_cli.py:35]: Jippie :)

File: tests/test_cli.py (hunk 3 above)
Comment (PicoCentauri) [reply tests/test_cli.py:62]: Uhh yes yours are also nice. We can adjust the if we find some edge cases.

**Comment by PicoCentauri** [comment]
I think the tests ara failing because 2.0 is not out yet...

**Comment by PicoCentauri** [comment]
Tests pass on my local machine with the current mda develop branch. So @joaomcteixeira you can merge when you are happy and ready.


Screenshot 2021-05-21 at 01 26 52

**Comment by joaomcteixeira** [comment]
Oh, you already did it here.
MDAnalysisTests are not needed. Sounds good. Let's merge!

File: tox.ini [hunk 4]
```diff
@@ -34,8 +34,8 @@ user_develop = false
 # by using tox the developer don't need to manage this dependencies
 # him/herself
 conda_deps =
-    MDAnalysis>=1
-    MDAnalysisTests
+    MDAnalysis==2.0.0b0
```
Comment (joaomcteixeira) [inline tox.ini:37]: wait. this is not good.

**Comment by joaomcteixeira** [comment]
@PicoCentauri 
Tests here have the same behaviour that in my computer. Can you check them to see if those that fail do not belong to this PR?

```
tests/test_cli.py::test_required_args PASSED                             [  3%]
tests/test_cli.py::test_wrong_module PASSED                              [  7%]
tests/test_cli.py::test_extra_options[version] PASSED                    [ 11%]
tests/test_cli.py::test_extra_options[help] PASSED                       [ 15%]
tests/test_cli.py::test_setup_clients[-s-topology-foo] PASSED            [ 19%]
tests/test_cli.py::test_setup_clients[-top-topology_format-foo] PASSED   [ 23%]
tests/test_cli.py::test_setup_clients[-f-trajectories-val2] PASSED       [ 26%]
tests/test_cli.py::test_setup_clients[-traj-trajectory_format-foo] PASSED [ 30%]
tests/test_cli.py::test_setup_clients[-atom_style-atom_style-foo] PASSED [ 34%]
tests/test_cli.py::test_setup_clients[-b-begin-42] PASSED                [ 38%]
tests/test_cli.py::test_setup_clients[-e-end-42] PASSED                  [ 42%]
tests/test_cli.py::test_setup_clients[-dt-dt-42] PASSED                  [ 46%]
tests/test_cli.py::Test_run_analsis::test_default FAILED                 [ 50%]
tests/test_cli.py::Test_run_analsis::test_topology_format FAILED         [ 53%]
tests/test_cli.py::Test_run_analsis::test_atom_style PASSED              [ 57%]
tests/test_cli.py::Test_run_analsis::test_trajectory_format FAILED       [ 61%]
tests/test_cli.py::Test_run_analsis::test_verbose FAILED                 [ 65%]
tests/test_cli.py::Test_run_analsis::test_traj_clices FAILED             [ 69%]
tests/test_save.py::test_get_cli_input PASSED                            [ 73%]
tests/test_utils.py::test_convert_str_time[1-1] PASSED                   [ 76%]
tests/test_utils.py::test_convert_str_time[-1--1] PASSED                 [ 80%]
tests/test_utils.py::test_convert_str_time[1ns-1000] PASSED              [ 84%]
tests/test_utils.py::test_convert_str_time[12e3-12000.0] PASSED          [ 88%]
tests/test_utils.py::test_convert_str_time[12e3ps-12000.0] PASSED        [ 92%]
tests/test_utils.py::test_convert_str_time_dt PASSED                     [ 96%]
tests/test_utils.py::test_convert_str_time_raise PASSED                  [100%]
```

**Comment by PicoCentauri** [comment]
> @PicoCentauri
> Tests here have the same behaviour that in my computer. Can you check them to see if those that fail do not belong to this PR?
> 
> ```
> tests/test_cli.py::test_required_args PASSED                             [  3%]
> tests/test_cli.py::test_wrong_module PASSED                              [  7%]
> tests/test_cli.py::test_extra_options[version] PASSED                    [ 11%]
> tests/test_cli.py::test_extra_options[help] PASSED                       [ 15%]
> tests/test_cli.py::test_setup_clients[-s-topology-foo] PASSED            [ 19%]
> tests/test_cli.py::test_setup_clients[-top-topology_format-foo] PASSED   [ 23%]
> tests/test_cli.py::test_setup_clients[-f-trajectories-val2] PASSED       [ 26%]
> tests/test_cli.py::test_setup_clients[-traj-trajectory_format-foo] PASSED [ 30%]
> tests/test_cli.py::test_setup_clients[-atom_style-atom_style-foo] PASSED [ 34%]
> tests/test_cli.py::test_setup_clients[-b-begin-42] PASSED                [ 38%]
> tests/test_cli.py::test_setup_clients[-e-end-42] PASSED                  [ 42%]
> tests/test_cli.py::test_setup_clients[-dt-dt-42] PASSED                  [ 46%]
> tests/test_cli.py::Test_run_analsis::test_default FAILED                 [ 50%]
> tests/test_cli.py::Test_run_analsis::test_topology_format FAILED         [ 53%]
> tests/test_cli.py::Test_run_analsis::test_atom_style PASSED              [ 57%]
> tests/test_cli.py::Test_run_analsis::test_trajectory_format FAILED       [ 61%]
> tests/test_cli.py::Test_run_analsis::test_verbose FAILED                 [ 65%]
> tests/test_cli.py::Test_run_analsis::test_traj_clices FAILED             [ 69%]
> tests/test_save.py::test_get_cli_input PASSED                            [ 73%]
> tests/test_utils.py::test_convert_str_time[1-1] PASSED                   [ 76%]
> tests/test_utils.py::test_convert_str_time[-1--1] PASSED                 [ 80%]
> tests/test_utils.py::test_convert_str_time[1ns-1000] PASSED              [ 84%]
> tests/test_utils.py::test_convert_str_time[12e3-12000.0] PASSED          [ 88%]
> tests/test_utils.py::test_convert_str_time[12e3ps-12000.0] PASSED        [ 92%]
> tests/test_utils.py::test_convert_str_time_dt PASSED                     [ 96%]
> tests/test_utils.py::test_convert_str_time_raise PASSED                  [100%]
> ```

They are related. I will fix them now.

File: tox.ini [hunk 5]
```diff
@@ -33,9 +33,10 @@ user_develop = false
 # installs dependencies we need for testing
 # by using tox the developer don't need to manage this dependencies
 # him/herself
+deps =
+    MDAnalysis==2.0.0b0
+    MDAnalysisTests==2.0.0b0
```
Comment (joaomcteixeira) [inline tox.ini:38]: add numpy here. also matplotlib

**Comment by PicoCentauri** [comment]
Tests are working now in Linux 🐧🚀! However, for windows 🪟 they fail

```
tests/test_utils.py::test_KwargsDict[-d {"key1":1}-expected0] FAILED     [ 93%]
```

There is some issue with file I/O on windows. I think this is something for another PR...

**Comment by joaomcteixeira** [comment]
The implementations here look interesting.
With a quick look I can't understand why tests were cancelled on MAC and Windows after py38 was success. Need to investigate this in another PR.
